app/Http/Controllers/BookingController.php

```diff
@@ -70,6 +70,56 @@ public function create(Request $request)
 
     }
 
+    public function report()
+    {
+        $rooms = Booking::select('rooms.id', 'rooms.name', 'bookings.start', 'bookings.end')
+            ->join('booking_room', 'bookings.id', '=', 'booking_room.booking_id')
+            ->join('rooms', 'booking_room.room_id', '=', 'rooms.id')
+            ->get();
+
+        $reportData = [];
+
+        foreach ($rooms as $room) {
+            $roomId = $room->id;
+            $roomName = $room->name;
+            $bookingStart = $room->start;
+            $bookingEnd = $room->end;
+
+            $startDate = new \DateTime($bookingStart);
+            $endDate = new \DateTime($bookingEnd);
+
+            $dateDiff = $startDate->diff($endDate);
+
+            if (!isset($reportData[$roomId])) {
+                $reportData[$roomId] = [
+                    'id' => $roomId,
+                    'name' => $roomName,
+                    'total_stay_duration' => 0,
+                    'booking_count' => 0,
+                ];
+            }
+
+            $reportData[$roomId]['total_stay_duration'] += $dateDiff->days;
+            $reportData[$roomId]['booking_count']++;
+        }
+
+        foreach ($reportData as &$roomData) {
+            $roomData['average_booking_duration'] = $roomData['booking_count'] > 0 ? round($roomData['total_stay_duration'] / $roomData['booking_count']) : 0;
```

Comment:
Could you update this call to round so it gives 1 decimal place? https://www.php.net/manual/en/function.round.php